arangod/Aql/QuerySnippet.cpp

```diff
@@ -678,16 +671,41 @@ auto QuerySnippet::prepareFirstBranch(
           if (myExp.size() > 1) {
             myExpFinal.insert({aqlCollection->name(), std::move(myExp)});
           }
+        } else {
+          if (localGraphNode->isDisjoint()) {
+            // the target server does not have anything to do with the particular
+            // collection (e.g. because the collection's shards are all on other
+            // servers), but it may be asked for this collection, because vertex
+            // collections are registered _globally_ with the TraversalNode and
+            // not on a per-target server basis.
+            // so in order to serve later lookups for this collection, we insert
+            // an empty string into the collection->shard map.
+            // on lookup, we will react to this.
+            localGraphNode->addCollectionToShard(aqlCollection->name(), "");
+          }
         }
       }
 
+      // TODO: We need to exclude DBServers which only do have Satellite
```

Comment:
aha, this may still be open.
We need to add specific tests for this.

Reply:
kanban created to clarify topic.